ssl/statem/statem_srvr.c

```diff
@@ -1953,8 +1952,7 @@ WORK_STATE tls_post_process_client_hello(SSL *s, WORK_STATE wst)
                 }
                 s->rwstate = SSL_NOTHING;
             }
-            cipher =
-                ssl3_choose_cipher(s, s->session->ciphers, SSL_get_ciphers(s));
+            cipher = s->choose_cipher_cb(s, s->session->ciphers);
```

Comment:
Here I would check if the cb is non-null and call it if so, otherwise call ssl3_choose_cipher.